templates/strings-structured.stencil

```diff
@@ -10,218 +10,55 @@ import Foundation
-            {% endfor %}
-            {% endif %}
-            {% if subenum5.subenums %}
-            {% for subenum6 in subenum5.subenums %}
-
-            case {{subenum6.name|swiftIdentifier|titlecase|snakeToCamelCase|escapeReservedKeywords}}({{subenum6.name|swiftIdentifier|titlecase|snakeToCamelCase}}{{enumName}})
-            enum {{subenum6.name|swiftIdentifier|titlecase|snakeToCamelCase}}{{enumName}} {
-              // maximum level reached, please make sure not to use keys with so many levels
-            }
-            {% endfor %}
-            {% endif %}
-          }
-          {% endfor %}
-          {% endif %}
-        }
-        {% endfor %}
-        {% endif %}
-      }
-      {% endfor %}
-      {% endif %}
-    }
-    {% endfor %}
-    {% endif %}
-  }
-  {% endfor %}
-  {% endif %}
+{% func enumBlock strings sp %}
+{{args.sp}}  {% for string in args.strings.strings %}
+{{args.sp}}  /// {{string.translation}}
+{{args.sp}}  case {{string.keytail|swiftIdentifier|titlecase|snakeToCamelCase|escapeReservedKeywords}}{% if string.params %}({{string.params.types|join}}){% endif %}
+{{args.sp}}  {% endfor %}
+{{args.sp}}  {% if args.strings.subenums %}
+{{args.sp}}  {% for subenum in args.strings.subenums %}
+
+{{args.sp}}  case {{subenum.name|swiftIdentifier|titlecase|snakeToCamelCase|escapeReservedKeywords}}({{subenum.name|swiftIdentifier|titlecase|snakeToCamelCase}}{{enumName}})
+{{args.sp}}  enum {{subenum.name|swiftIdentifier|titlecase|snakeToCamelCase}}{{enumName}} {
+{{args.sp}}    {% set sp %}{{args.sp}}  {% endset %}
+{{args.sp}}    {% call enumBlock subenum sp %}
+{{args.sp}}  }
+{{args.sp}}  {% endfor %}
+{{args.sp}}  {% endif %}
+{% endfunc %}
+  {% call enumBlock structuredStrings sp %}
 }
 
 extension {{enumName}}: CustomStringConvertible {
+{% func extensionBlock strings sp keySuffix %}
+{{args.sp}}  {% for string in args.strings.strings %}
+{{args.sp}}  {% if string.params %}
+{{args.sp}}  case .{{string.keytail|swiftIdentifier|titlecase|snakeToCamelCase|escapeReservedKeywords}}({{string.params.declarations|join}}):
+{{args.sp}}    return {{enumName}}.tr("{{string.key}}", {{string.params.names|join}})
+{{args.sp}}  {% else %}
+{{args.sp}}  case .{{string.keytail|swiftIdentifier|titlecase|snakeToCamelCase|escapeReservedKeywords}}:
+{{args.sp}}    return {{enumName}}.tr("{{string.key}}")
+{{args.sp}}  {% endif %}
+{{args.sp}}  {% endfor %}
+{{args.sp}}  {% if args.strings.subenums %}
+{{args.sp}}  {% for subenum in args.strings.subenums %}
+{{args.sp}}  {% set subkey %}levelSub{{keySuffix}}Key{{subenum.name|swiftIdentifier|titlecase|snakeToCamelCase|escapeReservedKeywords}}{% endset %}
+{{args.sp}}  case .{{subenum.name|swiftIdentifier|titlecase|snakeToCamelCase|escapeReservedKeywords}}(let {{subkey}}):
+
+{{args.sp}}    switch {{subkey}} {
+{{args.sp}}    {% set sp %}{{args.sp}}    {% endset %}
+{{args.sp}}    {% set keySuffix %}{{keySuffix}}Sub{% endset %}
```

Comment:
We might want to change that to `%{ set keySuffix %}{{keySuffix + 1}}{% endset %}` but iirc that requires a change in Stencil so that the `+` operator tries to _convert_ its parameters to `Number` (instead of failing if they are of type `String` without trying to interpret them as numbers)?